### PR TITLE
add nft apis to kusama + polkadot asset hub runtimes

### PR DESCRIPTION
**Pull request by gilescope**
Now that the NFT pallet is released we should also release the NFT runtime APIs.
Tasks:

- [x] use associated types rather than concrete types where possible.

### CODE REVIEW

**Comment by paritytech-cicd-pr** [comment]
The CI pipeline was cancelled due to failure one of the required jobs.
**Job name**: cargo-clippy
**Logs**: https://gitlab.parity.io/parity/mirrors/polkadot-sdk/-/jobs/3604769

**Review by jsidorenko — APPROVED** [review]

**Review by georgepisaltu — APPROVED** [review]

**Comment by gilescope** [comment]
(This needs to land so that the constants are exposed before the corollary can be landed in the runtime repo.)

**Comment by jsidorenko** [comment]
It would be ideal to wait for this PR to get merged first: https://github.com/paritytech/polkadot-sdk/pull/1914
In that case, we won't break the public API

**Comment by gilescope** [comment]
Happy to put a note on the runtime PR that we should not merge it till the PR you mentioned has landed. Agreed we want to minimise breaking changes.

**Comment by jsidorenko** [comment]
sounds good!

**Comment by gilescope** [comment]
Ok those changes are all incorporated now.

File: cumulus/parachains/runtimes/assets/common/src/lib.rs
Comment (joepetrowski) [inline cumulus/parachains/runtimes/assets/common/src/lib.rs:96]:
```suggestion
/// Type to identify a particular NFT collection.
```

File: cumulus/parachains/runtimes/assets/common/src/lib.rs
Comment (joepetrowski) [inline cumulus/parachains/runtimes/assets/common/src/lib.rs:99]:
```suggestion
/// Type to identify a particular NFT in a collection.
```

**Review by joepetrowski — COMMENTED** [review]
This should have a Fellowship PR for Polkadot and Kusama.

**Pull request closed without merging**
